controller/src/main/java/io/pravega/controller/store/stream/AbstractStreamMetadataStore.java

```diff
@@ -494,7 +494,13 @@ public OperationContext createContext(String scope, String name) {
     @Override
     public CompletableFuture<Void> completeRollingTxn(String scope, String name, Map<Long, Long> sealedActiveEpochSegments,
                                                       VersionedMetadata<CommittingTransactionsRecord> record, OperationContext context, Executor executor) {
-        return withCompletion(getStream(scope, name, context).completeRollingTxn(sealedActiveEpochSegments, record), executor);
+
+        CompletableFuture<Void> future = withCompletion(getStream(scope, name, context).completeRollingTxn(sealedActiveEpochSegments, record), executor);
+
+        future.thenCompose(result -> findNumSplitsMerges(scope, name, context, executor).thenAccept(simpleEntry ->
```

Comment:
It looks like that this code is the same as in _scaleSegmentsSealed_
what about creating a common method ?

I am still new with this code, maybe @shrids can give more hints

Reply:
No, scaleSegmentSealed updates activeSegment in addition to what rollingTxn does here.
If we create a common method, then we have to use if condition. In the end we may have one or two more lines of codes.